### PR TITLE
Updating ACOM, new phase mapping module code, some fixes

### PR DESCRIPTION
**Pull request by cophus**
This PR adds phase mapping for ACOM Crystal / OrientationMap classes.  It also includes an overhaul of the ACOM matching so that the kernel size is not applied to both the library and experimental data, but rather just the library.  Other small fixes also included.

### CODE REVIEW

File: py4DSTEM/process/diffraction/crystal.py [hunk 1]
```diff
@@ -779,17 +807,27 @@ def generate_diffraction_pattern(
 
         # Calculate excitation errors
         if foil_normal is None:
-            sg = self.excitation_errors(g)
+            sg = self.excitation_errors(
+                g,
+                precession_angle_degrees=precession_angle_degrees,
+            )
```
Comment (sezelt) [inline py4DSTEM/process/diffraction/crystal.py:813]: User should be able to specify number of precession steps

File: py4DSTEM/process/diffraction/crystal.py [hunk 2]
```diff
@@ -779,17 +807,27 @@ def generate_diffraction_pattern(
...
             foil_normal = (
                 orientation_matrix.T
                 @ (-1 * foil_normal[:, None] / np.linalg.norm(foil_normal))
             ).ravel()
-            sg = self.excitation_errors(g, foil_normal)
+            sg = self.excitation_errors(
+                g,
+                foil_normal=foil_normal,
+                precession_angle_degrees=precession_angle_degrees,
+            )
```
Comment (sezelt) [inline py4DSTEM/process/diffraction/crystal.py:823]: Allow setting precession steps

File: py4DSTEM/process/diffraction/crystal_ACOM.py [hunk 3]
```diff
@@ -27,8 +27,11 @@ def orientation_plan(
     angle_step_in_plane: float = 2.0,
     accel_voltage: float = 300e3,
     corr_kernel_size: float = 0.08,
-    radial_power: float = 1.0,
-    intensity_power: float = 0.25,  # New default intensity power scaling
+    sigma_excitation_error: float = 0.02,
+    precession_angle_degrees=None,
+    power_radial: float = 1.0,
+    power_intensity: float = 0.25,
+    power_intensity_experiment: float = 0.25,
```
Comment (sezelt) [inline py4DSTEM/process/diffraction/crystal_ACOM.py:34]: Precession steps should probably be able to be set here as well

File: py4DSTEM/process/diffraction/crystal_phase.py [hunk 4]
```diff
@@ -3,6 +3,7 @@
 from scipy.optimize import nnls
 import matplotlib as mpl
 import matplotlib.pyplot as plt
+from scipy.ndimage import gaussian_filter
 
 from emdfile import tqdmnd, PointListArray
 from py4DSTEM.visualize import show, show_image_grid
```
Comment (sezelt) [inline py4DSTEM/process/diffraction/crystal_phase.py:9]: Class names really shouldn't have underscores in them... I'm not sure why we didn't catch this earlier haha

File: py4DSTEM/process/diffraction/crystal_phase.py [hunk 5]
```diff
@@ -33,306 +35,1366 @@ def __init__(
...
-                cm(1.0 * i / len(self.orientation_maps))
-                for i in range(len(self.orientation_maps))
-            ]
 
-        fig, (ax) = plt.subplots(figsize=(6, 6))
-        ax.matshow(
-            np.zeros((self.orientation_maps[0].num_x, self.orientation_maps[0].num_y)),
-            cmap="gray",
+        # tolerance
+        tol2 = 1e-6
```
Comment (sezelt) [inline py4DSTEM/process/diffraction/crystal_phase.py:198]: This could have a better name

File: py4DSTEM/process/diffraction/crystal_phase.py [hunk 6]
```diff
@@ -33,306 +35,1366 @@ def __init__(
...
+                )
+                if a0 == ind_max:
+                    print("\033[1m" + line + "\033[0m")
+                else:
+                    print(line)
+            print("----------------------------")
+            line = "{:>12} {:>15}".format(f"{sum(phase_weights):.2f}", "fit total")
+            print("\033[1m" + line + "\033[0m")
+            line = "{:>12} {:>15}".format(f"{phase_residual:.2f}", "fit residual")
+            print(line)
```
Comment (sezelt) [inline py4DSTEM/process/diffraction/crystal_phase.py:585]: This isn't necessary to change, but I'd just point out the `tabulate` package, which is excellent for this kind of thing: https://github.com/astanin/python-tabulate

**Review by sezelt — COMMENTED** [review]
Overall looks good, I just pointed out a few minor things

File: py4DSTEM/process/diffraction/crystal.py (hunk 1 above)
Comment (cophus) [reply py4DSTEM/process/diffraction/crystal.py:813]: I'm planning to replace this with an analytic function, so my preference is to leave this as is for now.

File: py4DSTEM/process/diffraction/crystal.py (hunk 2 above)
Comment (cophus) [reply py4DSTEM/process/diffraction/crystal.py:823]: Same comment as previous.

File: py4DSTEM/process/diffraction/crystal_ACOM.py (hunk 3 above)
Comment (cophus) [reply py4DSTEM/process/diffraction/crystal_ACOM.py:34]: Same comment as previous.

File: py4DSTEM/process/diffraction/crystal_phase.py (hunk 4 above)
Comment (cophus) [reply py4DSTEM/process/diffraction/crystal_phase.py:9]: Good catch haha!

File: py4DSTEM/process/diffraction/crystal_phase.py (hunk 5 above)
Comment (cophus) [reply py4DSTEM/process/diffraction/crystal_phase.py:198]: updated to be tolerance_origin_2, explanatory comment added too.

File: py4DSTEM/process/diffraction/crystal_phase.py (hunk 6 above)
Comment (cophus) [reply py4DSTEM/process/diffraction/crystal_phase.py:585]: This was very tedious to implement, next time I will def check out this package!

**Review by sezelt — APPROVED** [review]
I'm satisfied

**Pull request merged**